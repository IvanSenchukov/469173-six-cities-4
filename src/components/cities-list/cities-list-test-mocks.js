export const mocks = {
  offers: [
    {
      id: `0`,
      name: `Wood and stone place`,
      type: `Private room`,
      reference: `#`,
      imageSrc: `img/room.jpg`,
      price: 80,
      period: `night`,
      addedToBookmarks: true,
      premium: false,
      rating: 4,
      city: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198]
    },
    {
      id: `1`,
      name: `Canal View Prinsengracht`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-02.jpg`,
      price: 132,
      period: `night`,
      addedToBookmarks: false,
      premium: false,
      rating: 4,
      city: `Amsterdam`,
      coordinates: [52.369553943508, 4.85309666406198]
    },
    {
      id: `2`,
      name: `Some really good offer`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-01.jpg`,
      price: 100,
      period: `night`,
      addedToBookmarks: true,
      premium: true,
      rating: 5,
      city: `Amsterdam`,
      coordinates: [52.3909553943508, 4.929309666406198]
    },
    {
      id: `3`,
      name: `Nice, cozy, warm big bed apartment`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-03.jpg`,
      price: 180,
      period: `night`,
      addedToBookmarks: false,
      premium: true,
      rating: 5,
      city: `Amsterdam`,
      coordinates: [52.3809553943508, 4.939309666406198]
    },
    {
      id: `4`,
      name: `Some good Paris apartments`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-03.jpg`,
      price: 250,
      period: `night`,
      addedToBookmarks: false,
      premium: true,
      rating: 5,
      city: `Paris`,
      coordinates: [48.8645266, 2.3494559]
    },
  ],
  cities: [
    {
      name: `Paris`,
      coordinates: [48.85341, 2.3488]
    },
    {
      name: `Cologne`,
      coordinates: [50.93333, 6.95]
    },
    {
      name: `Brussels`,
      coordinates: [50.85045, 4.34878]
    },
    {
      name: `Amsterdam`,
      coordinates: [52.38333, 4.9]
    },
    {
      name: `Hamburg`,
      coordinates: [53.57532, 10.01534]
    },
    {
      name: `Dusseldorf`,
      coordinates: [51.22172, 6.77616]
    },
    {
      name: `Aalborg`,
      coordinates: [57.046707, 9.935932]
    },
  ],
  offersForAllCities: [
    {
      id: `0`,
      name: `Wood and stone place`,
      type: `Private room`,
      reference: `#`,
      imageSrc: `img/room.jpg`,
      price: 80,
      period: `night`,
      addedToBookmarks: true,
      premium: false,
      rating: 4,
      city: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198]
    },
    {
      id: `1`,
      name: `Canal View Prinsengracht`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-02.jpg`,
      price: 132,
      period: `night`,
      addedToBookmarks: false,
      premium: false,
      rating: 4,
      city: `Cologne`,
      coordinates: [52.369553943508, 4.85309666406198]
    },
    {
      id: `2`,
      name: `Some really good offer`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-01.jpg`,
      price: 100,
      period: `night`,
      addedToBookmarks: true,
      premium: true,
      rating: 5,
      city: `Brussels`,
      coordinates: [52.3909553943508, 4.929309666406198]
    },
    {
      id: `3`,
      name: `Nice, cozy, warm big bed apartment`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-03.jpg`,
      price: 180,
      period: `night`,
      addedToBookmarks: false,
      premium: true,
      rating: 5,
      city: `Hamburg`,
      coordinates: [52.3809553943508, 4.939309666406198]
    },
    {
      id: `4`,
      name: `Some good Paris apartments`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-03.jpg`,
      price: 250,
      period: `night`,
      addedToBookmarks: false,
      premium: true,
      rating: 5,
      city: `Paris`,
      coordinates: [48.8645266, 2.3494559]
    },
    {
      id: `5`,
      name: `Some good Dusseldorf apartments`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-03.jpg`,
      price: 250,
      period: `night`,
      addedToBookmarks: false,
      premium: true,
      rating: 5,
      city: `Dusseldorf`,
      coordinates: [48.8645266, 2.3494559]
    },
    {
      id: `6`,
      name: `Some good Aalborg apartments`,
      type: `Apartment`,
      reference: `#`,
      imageSrc: `img/apartment-03.jpg`,
      price: 250,
      period: `night`,
      addedToBookmarks: false,
      premium: true,
      rating: 5,
      city: `Aalborg`,
      coordinates: [48.8645266, 2.3494559]
    },
  ]
};
