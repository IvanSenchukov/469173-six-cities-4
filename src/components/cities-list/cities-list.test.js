import React from "react";
import renderer from "react-test-renderer";
import CitiesList from "./cities-list.jsx";
import {mocks} from "./cities-list-test-mocks";

describe(`CitiesList Snapshots`, () => {
  it(`should renders correctly with two available cities and selected city`, function () {
    const tree = renderer
      .create(<CitiesList selectedCity={`Paris`} offers={mocks.offers} cities={mocks.cities} onCityClick={jest.fn()}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`should renders correctly with six available cities and selected city`, function () {
    const tree = renderer
      .create(<CitiesList selectedCity={`Paris`} offers={mocks.offersForAllCities} cities={mocks.cities} onCityClick={jest.fn()}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
