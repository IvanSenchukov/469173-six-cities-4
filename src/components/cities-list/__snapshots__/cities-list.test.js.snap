// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList Snapshots should renders correctly with six available cities and selected city 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          onClick={[Function]}
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          onClick={[Function]}
        >
          <span>
            Cologne
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          onClick={[Function]}
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          onClick={[Function]}
        >
          <span>
            Hamburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          onClick={[Function]}
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`CitiesList Snapshots should renders correctly with two available cities and selected city 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          onClick={[Function]}
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
